yarn-project/telemetry-client/src/start.ts

```diff
@@ -7,13 +7,27 @@ import { type TelemetryClient } from './telemetry.js';
 
 export * from './config.js';
 
-export async function createAndStartTelemetryClient(config: TelemetryClientConfig): Promise<TelemetryClient> {
+let initialised = false;
+let telemetry: TelemetryClient = new NoopTelemetryClient();
+
+export async function initTelemetryClient(config: TelemetryClientConfig): Promise<TelemetryClient> {
   const log = createLogger('telemetry:client');
+  if (initialised) {
+    log.warn('Telemetry client has already been initialized once');
+    return telemetry;
+  }
```

Comment:
Heads up you got a race condition if `OpenTelemetryClient.createAndStart` takes too long, and you call into `initTelemetryClient` again while it's initializing - you'd end up with a noop telemetry client instead of the otel one.

Though this should rarely happen since you're calling `init` effectively only once.